src/Core/Injector/Injector.php

```diff
@@ -527,7 +527,13 @@ public function convertServiceProperty($value)
 
         // Evaluate constants surrounded by back ticks
         if (preg_match('/^`(?<name>[^`]+)`$/', $value, $matches)) {
-            $value = defined($matches['name']) ? constant($matches['name']) : null;
+            if (defined($matches['name'])) {
+                $value = constant($matches['name']);
+            } elseif (getenv($matches['name'])) {
```

Comment:
I also think that env vars should take precedence over constants, so put that code first.